app/Http/Controllers/ThirdParty/CallPowerController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Chompy\Http\Controllers\ThirdParty;
+
+use Illuminate\Http\Request;
+use Chompy\Http\Controllers\Controller;
+use Chompy\Jobs\CreateCallPowerPostInRogue;
+
+class CallPowerController extends Controller
+{
+
+    /**
+     * Create a controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+    }
+
+    /**
+     *
+     * @param Request $request
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+            'mobile' => 'required',
+            'callpower_campaign_id' => 'required|integer',
+            'status' => 'required|string',
+            'call_timestamp' => 'required|date',
+            'call_duration' => 'required|integer',
+            'campaign_target_name' => 'required|string',
+            'campaign_target_title' => 'required|string',
+            'campaign_target_district' => 'required|string',
+            'callpower_campaign_name' => 'required|string',
+            'number_dialed_into' => 'required',
+        ]);
+
+        // Send to queued job.
+        CreateCallPowerPostInRogue::dispatch($request);
```

Comment:
Serializing the entire HTTP request in to the queue job feels a little odd to me, since the actual "request" will be done by the time the job picks it up. Instead, I wonder if we should grab the validated array & pass that along?

```php
$parameters = $request->validate([
    // ...
]);

CreateCallPowerPostInRogue::dispatch($parameters);
```

Reply:
ah yeah I thought this was a little weird too - this is a lot better! Updated :)